src/response_sender.cc

```diff
@@ -259,11 +259,19 @@ ResponseSender::IsCancelled()
   return pb_cancel_->IsCancelled();
 }
 
+bool
+ResponseSender::IsClosed()
+{
+  std::lock_guard<std::mutex> lk(mu_);
+  return closed_;
+}
+
 void
 ResponseSender::Close()
 {
   std::lock_guard<std::mutex> lk(mu_);
   closed_ = true;
+  response_factory_deleted_.exchange(true);
```

Comment:
why do we need `exchange(true)` here?

Reply:
`response_sender->Close()` is called [here](https://github.com/triton-inference-server/python_backend/pull/381/files#diff-f2bb64cbc9d4dc5d938ce965813b23178f53732e02e8c3a2afa707e287e92e57R751) when there's an error occurred. I think since we let the backend to clean up the response factory when there's an error, `response_factory_deleted_` should be set to true so that the response sender will not send the clean up message in the destructor.